v2/protocol/http/protocol.go

```diff
@@ -223,7 +223,6 @@ type msgErr struct {
 func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	m := NewMessageFromHttpRequest(req)
 	if m == nil || m.ReadEncoding() == binding.EncodingUnknown {
-		http.Error(rw, fmt.Sprintf("Cannot read CloudEvent: %s", binding.ErrUnknownEncoding), http.StatusBadRequest)
 		p.incoming <- msgErr{msg: nil, err: binding.ErrUnknownEncoding}
 		return // if there was no message, return.
```

Comment:
we should set the rw.Status to 415 as you are saying

Reply:
I don't think so @n3wscott, what about an empty request?

Reply:
that's why imo this shouldn't fail, even better not here in the protocol where a protocol user could decide to handle the encoding unknown case

Reply:
done

Reply:
slinky has a good point, we use to have a converter feature that would allow us to try to convert or wrap a bad message into a cloudevent and then deliver that to a consumer, how could we allow that use case and still do this if no other option was available?